pkg/recipes/driver/gitconfig.go

```diff
@@ -60,26 +63,35 @@ func getURLConfigKeyValue(secrets v20231001preview.SecretStoresClientListSecrets
 
 	// git config key will be in the format of url.<git url with secret details>.insteadOf
 	// and value returned will the original git url domain, e.g github.com
-	return fmt.Sprintf("url.%s.insteadOf", path), url.Hostname(), nil
+	return fmt.Sprintf("url.%s.insteadOf", path), fmt.Sprintf("%s://%s", url.Scheme, url.Hostname()), nil
 }
 
-// Updates the global Git configuration with credentials for a recipe template path and prefixes the path with environment, application, and resource name to make the entry unique to each recipe execution operation.
+// Updates the local Git configuration in terraform working directory with credentials for a recipe template path and prefixes the path with environment, application, and resource name to make the entry unique to each recipe execution operation.
 //
 // Retrieves the git credentials from the provided secrets object
 // and adds them to the Git config by running
-// git config --global url<template_path_domain_with_credentails>.insteadOf <template_path_domain>.
-func addSecretsToGitConfig(secrets v20231001preview.SecretStoresClientListSecretsResponse, recipeMetadata *recipes.ResourceMetadata, templatePath string) error {
+// git config --file .git/config url<template_path_domain_with_credentails>.insteadOf <template_path_domain>.
+func addSecretsToGitConfig(workingDirectory string, secrets v20231001preview.SecretStoresClientListSecretsResponse, templatePath string) error {
+	if !strings.HasPrefix(templatePath, "git::") || reflect.DeepEqual(secrets, v20231001preview.SecretStoresClientListSecretsResponse{}) {
+		return nil
+	}
+	// Initialize a new Git repository in the terraform working directory.
+	_, err := git.PlainInit(workingDirectory, false)
```

Comment:
Can this working directory be existing before we initialize it by any chance?

Reply:
yes! its is created before we initialize it